app/models/legal_aid_application.rb

```diff
@@ -157,6 +168,53 @@ def ccms_case_reference
     most_recent_ccms_submission.case_ccms_reference
   end
 
+  def add_default_substantive_scope_limitation!
+    ApplicationScopeLimitation.create!(
+      legal_aid_application: self,
+      scope_limitation: default_substantive_scope_limitation_for_first_proceeding_type,
+      substantive: true
+    )
+  end
+
+  def add_default_delegated_functions_scope_limitation!
+    ApplicationScopeLimitation.create!(
+      legal_aid_application: self,
+      scope_limitation: default_delegated_functions_scope_limitation_for_first_proceeding_type,
+      substantive: false
+    )
+  end
+
+  def default_substantive_scope_limitation_for_first_proceeding_type
+    lead_proceeding_type.default_substantive_scope_limitation
+  end
+
+  def default_delegated_functions_scope_limitation_for_first_proceeding_type
+    lead_proceeding_type.default_delegated_functions_scope_limitation
+  end
+
+  def reset_delegated_functions
+    self.used_delegated_functions = false
+    self.used_delegated_functions_on = nil
+  end
+
+  def reset_proceeding_types!
+    proceeding_types.clear
+    scope_limitations.clear
+    reset_delegated_functions
+  end
+
+  def default_cost_limitation
+    used_delegated_functions? ? default_delegated_functions_cost_limitation : default_substantive_cost_limitation
+  end
+
+  def default_substantive_cost_limitation
+    lead_proceeding_type.default_cost_limitation_substantive
+  end
+
+  def default_delegated_functions_cost_limitation
+    lead_proceeding_type.default_cost_limitation_delegated_functions
+  end
```

Comment:
I think `default_substantive_scope_limitation`, `default_delegated_functions_scope_limitation`, `default_cost_limitation_substantive`, and `default_cost_limitation_delegated_functions` should all be delegated to `lead_proceeding_type`. In that way, it is made clear that construction of those values belongs to the `lead_proceeding_type`. It also makes it simpler to ensure that they are all handled the same way, rather than having two of them suffixed with `_for_first_proceeding_type` and two of them not.

Reply:
I've removed the methods `default_substantive_scope_limitation_for_first_proceeding_type` and `default_delegated_functions_scope_limitation_for_first_proceeding_type` and replaced the lines that referenced them with `lead_proceeding_type.default_substantive_scope_limitation` and `lead_proceeding_type.default_delegated_functions_scope_limitation`.  This makes it absolutely clear where it's coming from.  Just calling it `default ....` and delegating it to the lead_proceeding_type which cause more pain when we come to add in multiple proceeding types, I feel.